src/rgw/store/simplefile/bucket.cc

```diff
@@ -0,0 +1,386 @@
+    ceph_assert(std::filesystem::exists(data_path));
+    ceph_assert(std::filesystem::is_regular_file(data_path));
+
+    auto obj = get_object(objkey);
+    rgw_bucket_dir_entry dirent;
+    dirent.key = cls_rgw_obj_key(name);
+    dirent.meta.accounted_size = obj->get_obj_size();
+    dirent.meta.mtime = obj->get_mtime();
+    results.objs.push_back(dirent);
+  }
+
+  /*
+  for (auto const &dir_entry :
+       std::filesystem::directory_iterator{objects_path()}) {
+    if (!dir_entry.is_regular_file()) {
+      lsfs_dout(dpp, 10) << "skipping " << dir_entry << dendl;
+      continue;
+    }
+
+    auto fn = dir_entry.path().filename();
+    if (fn.c_str()[0] == '_') {
+      lsfs_dout(dpp, 10) << "skipping meta entry " << fn << dendl;
+      continue;
+    }
+
+    lsfs_dout(dpp, 10) << "adding " << fn << dendl;
+    auto obj = get_object(rgw_obj_key(fn.string()));
+
+    rgw_bucket_dir_entry entry;
+    entry.key = cls_rgw_obj_key(fn.string());
+    entry.meta.accounted_size = obj->get_obj_size();
+    entry.meta.mtime = obj->get_mtime();
+    results.objs.push_back(entry);
+  }
+  */
+  lsfs_dout(dpp, 10) << "found " << results.objs.size() << " objects" << dendl;
+  return 0;
+}
+
+int SimpleFileBucket::remove_bucket(const DoutPrefixProvider *dpp,
+                                    bool delete_children,
+                                    bool forward_to_master, req_info *req_info,
+                                    optional_yield y) {
+  /** Remove this bucket from the backing store */
+  ldpp_dout(dpp, 10) << __func__ << ": TODO" << dendl;
+  return -ENOTSUP;
+}
+
+int SimpleFileBucket::remove_bucket_bypass_gc(int concurrent_max,
+                                              bool keep_index_consistent,
+                                              optional_yield y,
+                                              const DoutPrefixProvider *dpp) {
+  /** Remove this bucket, bypassing garbage collection.  May be removed */
+  ldpp_dout(dpp, 10) << __func__ << ": TODO" << dendl;
+  return -ENOTSUP;
+}
+
+int SimpleFileBucket::load_bucket(const DoutPrefixProvider *dpp,
+                                  optional_yield y, bool get_stats) {
+  std::filesystem::path meta_file_path = bucket_metadata_path();
+  ceph_assert(std::filesystem::exists(meta_file_path));
+  JSONParser bucket_meta_parser;
+  if (!bucket_meta_parser.parse(meta_file_path.c_str())) {
+    lsfs_dout(dpp, 10) << "Failed to parse bucket metadata from "
+                       << meta_file_path << ". Returing EINVAL" << dendl;
+    return -EINVAL;
+  }
+
+  auto it = bucket_meta_parser.find("meta");
+  ceph_assert(!it.end());
+
+  SimpleFileBucket::Meta meta;
+  JSONDecoder::decode_json("meta", meta, &bucket_meta_parser); 
+  lsfs_dout(dpp, 10) "bucket name: " << meta.info.bucket.get_key() << dendl;
+
+  info = meta.info;
+  // multipart = meta.multipart;
+
+  load_object_map(dpp);
+
+  auto f = new JSONFormatter(true);
```

Comment:
Does it make sense to use this code? This way you do not need to take care to delete the new instantiated object.
```suggestion
  JSONFormatter f(true);
```